x-pack/legacy/plugins/epm/public/constants.ts

```diff
@@ -6,8 +6,11 @@
 import { IconType } from '@elastic/eui';
 import { ElasticsearchAssetType, KibanaAssetType, AssetType, ServiceName } from '../common/types';
 
-// TODO: figure out how to allow only corresponding asset types (KibanaAssetType, ElasticsearchAssetType)
-export const DisplayedAssets: Record<ServiceName, AssetType[]> = {
+// only allow Kibana assets for the kibana key, ES asssets for elasticsearch, etc
+type ServiceNameToAssetTypes = Record<Extract<'kibana', ServiceName>, KibanaAssetType[]> &
+  Record<Extract<'elasticsearch', ServiceName>, ElasticsearchAssetType[]>;
+
+export const DisplayedAssets: ServiceNameToAssetTypes = {
```

Comment:
Isn't this similar to doing:


```
interface DisplayedAssets{
  kibana: KibanaAssetType[],
  elasticsearch: ElasticsearchAssetType[],
}

```

Reply:
Yes it's similar. I think the key difference is that an interface doesn't have any type info or restrictions on the key. You could enter `foo` and it would not complain. Also, consumers would only know that the key was a string or number, now they know exactly what the type of the key is.